### PR TITLE
refactored profile view

### LINKED ISSUE #438: Refactor profile page
## ✨ Feature request
Use compositions api instead of options api.

### 📜 Acceptance Criteria
- [ ] Profile view has been refactored.
- [ ] Components dependent on profile has been refactored.

### 💡 Additional context
- AddLoan will be refactored in #443

### PR DESCRIPTION
**Pull request by TE4-Filip-Liljenberg**
## ⚙️ What issue did you implement or fix?
Fixes #438

## 📜 Summary
- Refactored profile view

## 📝 Type of change

Please delete options that are not relevant.

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by te4-adrian-almetun-smeds — APPROVED** [review]
Looks good :+1:
Just check the failing tests

**Comment by te4-david-jensen** [comment]
why remove type ```UsersModuleType```?

**Review by Te4-Jacob-sandstrom — COMMENTED** [review]
ERROR in /home/runner/work/Yabs/Yabs/frontend/src/views/__tests__/ProfileView.spec.ts(29,30):
29:30 'ProfileView' refers to a value, but is being used as a type here.
    27 |   it('Displays the correct name/role/class', () => {
    28 |     setTimeout(() => {
  > 29 |       const wrapper: Wrapper<ProfileView> = shallowFactory(ProfileView);
       |                              ^
    30 |   
    31 |       baseUser.role = 1;
    32 |       baseUser.uid = 1111;

change  Wrapper<ProfileView> to  Wrapper<Vue> in test file

File: frontend/src/types.ts [hunk 1]
```diff
@@ -104,11 +103,4 @@ interface LoanCollection {[id: number]: Loan; }
 interface Subject {
   id: number;
   name: string;
-}
-
-interface UserModuleType {
-  currentUser: User;
-  currentUserID: number;
-  all: UserCollection;
-  allAsArray: User[];
```
Comment (te4-david-jensen) [inline frontend/src/types.ts:113]: Why remove this type?

**Review by te4-david-jensen — CHANGES_REQUESTED** [review]
Just make the final checks pass and it's good to go! (message me on discord if you want help)

**Review by te4-david-jensen — APPROVED** [review]
Looks good! :shipit:

**Pull request merged**